### PR TITLE
[UT] add skiplist for PVC w/ rolling driver

### PR DESCRIPTION
**Pull request by AshburnLee**
## Functionality
Renamed default to pvc_rolling as pvc with rolling is the default platform.
Request: https://github.com/intel/intel-xpu-backend-for-triton/issues/985
~~This PR add a skiplist for pytest enables skipping ‘SKIPPED’ cases in `script/skiplist/pvc_rolling` when the platform is **PVC** with **Rolling** driver.~~

## Result
~~After the change of this PR, the test sessions with deselect files skip all the SKIPPED test cases.~~

### CODE REVIEW

File: scripts/capture-hw-details.sh [hunk 1]
```diff
@@ -50,3 +50,50 @@ if [ "$QUIET" = false ]; then
...
+agama_to_driver_type() {
+    local lts=(803)
+    local rolling=(821 775)
+
+    local agama=$1
+    local found="false"
+
+    for item in "${lts[@]}"; do
+        if [[ "$agama" -eq "$item" ]]; then
+            echo "ltd"
```
Comment (gshimansky) [inline scripts/capture-hw-details.sh:63]: echo "lts"

Comment (AshburnLee) [reply scripts/capture-hw-details.sh]: done, thanks~

File: scripts/pytest-utils.sh [hunk 2]
```diff
@@ -34,6 +40,12 @@ pytest() {
         mkdir -p "$CURRENT_SKIPLIST_DIR"
         # skip comments in the skiplist
         sed -e '/^#/d' "$TRITON_TEST_SKIPLIST_DIR/$TRITON_TEST_SUITE.txt" > "$CURRENT_SKIPLIST_DIR/$TRITON_TEST_SUITE.txt"
+
+        # put skiplist/pvc_{rolling/lts}/language.txt to the current skiplist
+        if [[ $TRITON_TEST_SUITE == "language" && ! -z $PVC_DRIVER_SKIPLIST_DIR ]]; then
+            sed -e '/^#/d' "$PVC_DRIVER_SKIPLIST_DIR/$TRITON_TEST_SUITE.txt" >> "$CURRENT_SKIPLIST_DIR/$TRITON_TEST_SUITE.txt"
+        fi
```
Comment (pbchekin) [inline scripts/pytest-utils.sh:47]: The skiplist can be changed by setting `TRITON_TEST_SKIPLIST_DIR`, no need to overwrite the exiting files in the current skiplist.

**Review by pbchekin — COMMENTED** [review]
I think we do not need to put the existing xfailed tests to the skiplist. We use skiplist for skipped tests instead. Also the default skiplist is for PVC with the rolling driver, no need to create a new one, instead consider renaming default to pvc-rolling.

File: scripts/capture-hw-details.sh [hunk 3]
```diff
@@ -50,3 +50,50 @@ if [ "$QUIET" = false ]; then
     echo "AGAMA_VERSION=$AGAMA_VERSION"
     echo "GPU_DEVICE=$GPU_DEVICE"
 fi
+
+agama_to_driver_type() {
+    local lts=(803)
+    local rolling=(821 775)
```
Comment (vlad-penkin) [inline scripts/capture-hw-details.sh:56]: Please do not hardcode the Rolling driver version numbers.

**Review by vlad-penkin — CHANGES_REQUESTED** [review]

**Comment by AshburnLee** [comment]
> I think we do not need to put the existing xfailed tests to the skiplist. We use skiplist for skipped tests instead. Also the default skiplist is for PVC with the rolling driver, no need to create a new one, instead consider renaming default to pvc-rolling.

Oh, I may understand it wrong, I should have put **skipped** tests in the skiplist, not the **xfailed**. OK, will change that and renaming default file.

How about the xfailed cases? do we just leave it as it is? thansks @pbchekin

**Comment by pbchekin** [comment]
> How about the xfailed cases? do we just leave it as it was?

Yes

File: scripts/capture-hw-details.sh (hunk 3 above)
Comment (AshburnLee) [reply scripts/capture-hw-details.sh]: Remove these two conversion functions as the default platform is PVC with Rolling.

File: scripts/pytest-utils.sh (hunk 2 above)
Comment (AshburnLee) [reply scripts/pytest-utils.sh]: Renamed `/default` to `/pvc_rolling`, removed `PVC_DRIVER_SKIPLIST_DIR`

**Comment by AshburnLee** [comment]
How about approving this PR? Just change the skip list default name as 'pvc_rolling'? @vlad-penkin

**Review by pbchekin — APPROVED** [review]

**Comment by pbchekin** [comment]
@AshburnLee please sync the PR with the target branch, there is a change to the skip list in #1310.

**Comment by AshburnLee** [comment]
> @AshburnLee please sync the PR with the target branch, there is a change to the skip list in #1310.

Done, thanks~

**Comment by vlad-penkin** [comment]
@AshburnLee similar comment, please take into the account this PR:
- #1319

**Comment by AshburnLee** [comment]
> @AshburnLee similar comment, please take into the account this PR:
> 
> * [[UT] Enable more convert2d tests #1319](https://github.com/intel/intel-xpu-backend-for-triton/pull/1319)

Updated, thanks~

**Comment by AshburnLee** [comment]
Please give approval @vlad-penkin thanks~

**Comment by pbchekin** [comment]
No longer needed.

**Pull request closed without merging**